packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -93,6 +96,13 @@ export const createPluginActions: CreatePluginActions = (
     onUnknownMessage,
   }
 
+  const onHeightChange = (height: number) => {
+    if (state.isModalOpen) {
+      return
+    }
+    postToContainer(heightChangeMessage(uid, `${height}px`))
```

Comment:
what if the user wants to set another unit ? such as rem or em? should we consider that case

Reply:
@BibiSebi, I think if we designed the modal feature from scratch, we would simply always set the height to `auto` in modal mode, and the height in pixels in non-modal mode (as I described above). I.e. in modal mode, the Visual Editor would simply set the height to `auto` and ignore all `HeightChangeMessage`s, while in non-modal mode, the Visual Editor would use the height in pixels from `HeightChangeMessage`.

The more I think of this, the more I think that using `auto` is a [leaky abstraction](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/). The field plugin should not have to know that it's embedded in a flexbox layout; it should just know that in modal mode, the height is constant-- while  in non-modal mode, the height is variable.

So to answer your question; probably not. `heightChangeMessage` should mean that the height of the field plugin changed, and the value in the payload should correspond to the height of the document. `rem` and `em` are units that are relative to the font size outside of the iframe, of which the plugin has no knowledge.

Reply:
![Screenshot 2023-06-29 at 17 47 38](https://github.com/storyblok/field-plugin/assets/499898/372f471b-bcd1-4d68-b641-6dee95c65145)

Oh, I left a comment, but... 😂